addons/mail/models/mail_activity.py

```diff
@@ -99,10 +122,14 @@ def _compute_state(self):
     @api.onchange('activity_type_id')
     def _onchange_activity_type_id(self):
         if self.activity_type_id:
-            if not self.summary:
-                self.summary = self.activity_type_id.summary
+            self.summary = self.activity_type_id.summary
```

Comment:
When you are going to change activity it will populate summary field automatically. let take one example, you have one activity having summary and another having without summary. so when you select 1st one summary will auto populate. after that if you select second one which does not have summary. it will not update summary field as it does not have value. so it will keep previous summary which is wrong, it needs to update it as blank. 

You can see this bug in master so it is part of it. let me know if you need more clarification for the same.